Synced node: java.lang.UnsupportedOperationException: empty.last on startup

**Describe the bug**

We fail with an exception if attempt to sync when we are already synced with the chain. I stopped my node, and then restarted it again immediately and received this error:

```
2020-04-09T11:35:09 INFO [HTTP] Started Bitcoin-S HTTP server at /127.0.0.1:9999
2020-04-09T11:35:14 INFO [P2P] Received headers message with 0 headers
[ERROR] [04/09/2020 06:35:14.939] [bitcoin-s-akka.actor.default-dispatcher-6] [akka://bitcoin-s/user/P2PClient$-1586432107225] empty.last
java.lang.UnsupportedOperationException: empty.last
        at scala.collection.immutable.Vector.last(Vector.scala:284)
        at org.bitcoins.server.Main$.$anonfun$createCallbacks$7(Main.scala:166)
        at org.bitcoins.node.NodeCallbacks.$anonfun$executeOnBlockHeadersReceivedCallbacks$2(NodeCallbacks.scala:94)
        at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:433)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:92)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)

```

--- a/app/server/src/main/scala/org/bitcoins/server/Main.scala
+++ b/app/server/src/main/scala/org/bitcoins/server/Main.scala
@@ -7,6 +7,7 @@ import akka.actor.ActorSystem
 import org.bitcoins.chain.config.ChainAppConfig
 import org.bitcoins.core.Core
 import org.bitcoins.core.api.ChainQueryApi
+import org.bitcoins.core.util.FutureUtil
 import org.bitcoins.db.AppConfig
 import org.bitcoins.keymanager.KeyManagerInitializeError
 import org.bitcoins.keymanager.bip39.BIP39KeyManager
@@ -163,7 +164,11 @@ object Main extends App {
       wallet.processBlock(block).map(_ => ())
     }
     lazy val onHeaders: OnBlockHeadersReceived = { headers =>
-      wallet.updateUtxoPendingStates(headers.last).map(_ => ())
+      if (headers.isEmpty) {
+        FutureUtil.unit
+      } else {
+        wallet.updateUtxoPendingStates(headers.last).map(_ => ())
+      }
     }
     if (nodeConf.isSPVEnabled) {
       Future.successful(

